### PR TITLE
Add support for @MockBean

### PR DESCRIPTION
**Pull request by geoand**
With this annotation, CDI beans will automatically be mocked using mockito.

**TODO**:

- [ ] Add support for qualifiers
- [ ] Add documentation

### CODE REVIEW

**Comment by geoand** [comment]
Adding `coding-in-progress` in order to account for feedback and the fact that if people think it's a good idea I also need to document it in the testing guide.

File: integration-tests/mockbean/src/test/java/io/quarkus/it/mockbean/GreetingResourceTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.mockito.Mockito;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.MockBean;
+
+@QuarkusTest
+class GreetingResourceTest {
+
+    @MockBean
+    MessageService messageService;
```
Comment (mkouba) [inline integration-tests/mockbean/src/test/java/io/quarkus/it/mockbean/GreetingResourceTest.java]: Currently, you can do:
```java
@Produces
@Mock
MessageService mockMessageService() {
   return Mockito.mock(MessageService.class);
}

@Inject
Message
```

Comment (geoand) [reply integration-tests/mockbean/src/test/java/io/quarkus/it/mockbean/GreetingResourceTest.java]: Yes absolutely!

The PR doesn't bring in fundamental capability that didn't already exist. It does however lighten the burden for users if they have multiple mocks they want to use in their tests (whether that is good or bad is another question :) )

**Comment by cescoffier** [comment]
@geoand @mkouba what's the status of this PR?

**Comment by geoand** [comment]
On hold for the time being

**Comment by gsmet** [comment]
I think a first good step would be to document what @mkouba is saying. Then we could see if it needs improvements.

Keep in mind that most users are not accustomed to CDI so this is not obvious.

**Comment by geoand** [comment]
I'm going to close this since it's very out of date. If needed in the future, we can revisit

**Pull request closed without merging**